torch/csrc/distributed/rpc/rref_context.cpp

```diff
@@ -390,11 +406,9 @@ void RRefContext::addSelfAsFork(c10::intrusive_ptr<OwnerRRef>& rref) {
 void RRefContext::addForkOfOwner(const RRefId& rrefId, const ForkId& forkId) {
```

Comment:
I wonder if we should split this function into two in this case. Because this function is used in both `RREF_FORK_REQUEST` and `SCRIPT/PYTHON_REMOTE_CALL`. We should make the former idempotent but as the latter contains user code, we cannot try them twice.

Reply:
Do you mean a separate function like `addForkOfOwnerIfNotPresent` (or something along those lines) - and we call this idempotent function from when `RREF_FORK_REQUEST` is received, and use the non-idempotent function otherwise? Alternately, is it alright to make the existing function idempotent in all cases, or do you specifically want it throw an exception for the case of a repeated fork for `SCRIPT/PYTHON_REMOTE_CALL` messages?

Reply:
Would be nice to have an idempotent variant only in cases where we want it to be idempotent. So we should split this into two functions and call the idempotent one only when required.